src/main.py

```diff
@@ -0,0 +1,170 @@
+
+    return questionnaires
+
+
+def load_filled_questionnaires(output_dir: Path) -> Dict[str, Dict[str, str]]:
+    """
+    Loads already (partially) filled in questionnaires from yaml files in the output_dir.
+
+    :param output_dir: Path to a directory containing question and answer yaml files.
+    :return: A dictionary mapping the name of a questionnaire to a dictionary with keys being
+    the question and the value being the answer.
+    :raises: TypeError: If output_dir is not a directory path.
+
+    """
+    if not Path.is_dir(output_dir):
+        raise TypeError(f"{output_dir} must be a directory")
+
+    q_and_a_sources = {}
+    for q_and_a_filepath in output_dir.iterdir():
+        if q_and_a_filepath.suffix != ".yaml":
+            print(f"ignoring unexpected file format {q_and_a_filepath.suffix}", file=sys.stderr)
+            continue
+
+        with open(q_and_a_filepath) as f:
+            q_and_a_yaml = yaml.safe_load(f)
+
+        q_and_a_sources[q_and_a_filepath.stem] = {}
+        for q_and_a in q_and_a_yaml:
+            q_and_a_sources[q_and_a_filepath.stem][q_and_a["question"]] = q_and_a["answer"]
+
+    return q_and_a_sources
+
+
+if __name__ == "__main__":
+    cwd = Path.cwd()
+    question_dir = (cwd / "questionnaires").resolve()
+    output_dir = (cwd / "out").resolve()
+
+    Path(output_dir).mkdir(parents=True, exist_ok=True)
+
+    print("=" * 50)
+    print("Welcome to TAD! We have a few questions for you.")
+    print("=" * 50)
+
+    questionnaires = load_questionnaires(question_dir)
+    questionnaires_with_answers = load_filled_questionnaires(output_dir)
+
+    user_aborted = False
+    for questionnaire in questionnaires:
+        output = {}
+        try:
+            # load answers from user
+            question_and_answers = None
+            if questionnaire.name in questionnaires_with_answers:
+                question_and_answers = questionnaires_with_answers[questionnaire.name]
+
+            for question in questionnaire.questions:
+                if question_and_answers is not None and question["name"] in question_and_answers:
+                    print(question["name"])
+
+                    if isinstance(question_and_answers[question["name"]], list):
+                        print("Previously given answer: " + ", ".join(question_and_answers[question["name"]]))
+                    else:
+                        print(f"Previously given answer: {question_and_answers[question["name"]]}")
+
+                    confirm = questionary.confirm(message="Keep this answer?").ask()
+                    if confirm:
+                        tmp = {question["name"]: question_and_answers[question["name"]]}
+                    else:
+                        tmp = questionary.unsafe_prompt(question)
+                else:
+                    tmp = questionary.unsafe_prompt(question)
+                output.update(tmp)
+        except KeyboardInterrupt:
+            user_aborted = True
```

Comment:
can you add a general execpt with logging?

Reply:
Added basic logging with standard logging library. Info log message is emitted when user aborts.